### PR TITLE
Fixes #20623 - Failed scenario should exit with code 1

### PR DESCRIPTION
**Pull request by swapab**

### CODE REVIEW

**Comment by swapab** [comment]
@iNecas https://travis-ci.org/theforeman/foreman_maintain/jobs/274487488
This is failing on my local as well. Do you know what might be causing the tags to show twice?

File: lib/foreman_maintain/runner.rb [hunk 1]
```diff
@@ -84,6 +86,14 @@ def storage
 
     private
 
+    def execute_scenario_steps(scenario)
+      scenario.before_scenarios.flatten.each { |before_scenario| run_scenario(before_scenario) }
+      yield
```
Comment (iNecas) [inline lib/foreman_maintain/runner.rb]: I think we can also move `return if quit? # the before scenarios caused the stop of the execution` here and not use the `yield`, for readability purposes

**Comment by iNecas** [comment]
Don't know. Please try to find the cause: we can discuss on status call today.

File: lib/foreman_maintain/runner.rb (hunk 1 above)
Comment (swapab) [reply lib/foreman_maintain/runner.rb]: Yes, I thought returning from extracted-method might cause some unexpected behaviour.

**Comment by swapab** [comment]
It was `require 'checks/dummy'` in `scenarios/dummy.rb`. :smile:
Removed.

**Comment by swapab** [comment]
Added `exit!` to exit with code `1` when encountered with `UsageError`.

File: lib/foreman_maintain/cli.rb [hunk 2]
```diff
@@ -19,19 +19,42 @@ def run(*arguments)
...
+        end
+
+        puts error.message
+        logger.error(error)
+
+        @exit_code = 1
+      end
+
+      def process_usage_error(error)
+        log_exit_code_info(exit_code)
```
Comment (iNecas) [inline lib/foreman_maintain/cli.rb]: I'm getting `undefined local variable or method `exit_code' for` when running `bin/foreman-maintain upgrade run --target-version nonexisting`

Comment (swapab) [reply lib/foreman_maintain/cli.rb]: That's a test case missed. Fixed!

**Comment by iNecas** [comment]
This branch has conflicts now, please rebsae

**Comment by swapab** [comment]
Conflicts resolved. Rebased.

**Comment by iNecas** [comment]
Looks good, thanks @swapab

**Pull request merged**